hydra-plutus/src/Hydra/Contract/OnChain.hs

```diff
@@ -136,20 +133,27 @@ hydraValidatorHash = Scripts.validatorHash . hydraTypedValidator
 
 data Initial
 instance Scripts.ValidatorTypes Initial where
-  type DatumType Initial = PubKeyHash
+  -- NOTE: Ideally, we would like to get rid of validator hashes in here,
+  -- because they are statically known!
+  -- So instead of passing them around, we could simply call
+  -- `hydraValidatorHash` and `commitValidatorHash` in the `initialValidator`...
+  --
+  -- However, this makes the compiler chokes with:
+  --
+  --     GHC Core to PLC plugin: E042:Error: Unsupported feature: Type constructor: GHC.Prim.ByteArray#
```

Comment:
This error looks familiar. It's from the plutus compiler when the data type has no `ToData` instance or so.. likely the `ValidatorHash` type is just a newtype wrapper around some `ByteArray` for which there is no `PlutusTx` equivalent. Maybe marshalling via `ByteString` works?